### DISCUSSION TITLE
error: no auth cookie provided.

### DISCUSSION THREAD

**Discussion by KreqHex** [FuseFairy/DiscordBot-EdgeGPT#129] [General]
I got a question.
Everythings works great but the only this that is not working is the /create image command.
when i do this i get the respons: error: no auth cookie provided.
How can i fix this?

i already tried some things but nothing works.
please help :)

**Comment by FuseFairy** [comment]
Make sure your cookie contains the "name": "_U" value

**Reply by KreqHex** [reply]
i did everything what is in the readme.
in my cookie i have the "name": "value",
so what do i wrong?

**Reply by KreqHex** [reply]
see screenshot for my cookie:

![screenshot](https://github.com/FuseFairy/DiscordBot-EdgeGPT/assets/119231745/811e7b1d-901f-47e9-ba3b-8d74df94270c)

**Reply by FuseFairy** [reply]
An error occurs when the name "_U" of the cookie has the wrong value. 
![image](https://github.com/FuseFairy/DiscordBot-EdgeGPT/assets/78064680/70e55a08-f963-45e3-b3d2-7467089be54a)

Go to the BIngImageCreator website and make sure you are licensed to create images, and try using a private page to get a cookie or clear the website cookie and then get the cookie again.

**Reply by KreqHex** [reply]
i did everything you ask.
but still doenst work.
now i got this error message from the bot:  Error: 'NoneType' object is not iterable

**Reply by FuseFairy** [reply]
So is the name `_U` in the cookies?

**Reply by KreqHex** [reply]
yes i changed it to _U.
but in my real cookies there is no _U.
But i am logged in with microsoft account on bing on the imagecreator website.
but cookie is still the same.
(you cant go in incognito it gives no cookie)

so i deleted all cookies and go again to the website en take new cookie.
but still doenst work.

**Reply by KreqHex** [reply]
Its free right?
you dont have a paid plan in bing or something to get good cookie?

**Reply by FuseFairy** [reply]
It's free, and you can't change your cookie, it doesn't make sense.

**Reply by FuseFairy** [reply]
Please make sure you are logged in before obtaining cookies.

**Reply by KreqHex** [reply]
Yes i am.
i am logged in before i get the cookie.
first i clean my browsers cookies.
then i go to the image creator website from bing.
then i log in to my account.
then i grap the cookies.
export it as JS
then paste it in the cookies.js file.
and then i start the bot.
and then its not working.

**Reply by FuseFairy** [reply]
I can see why you can't run the bot, you need to paste all the cookies into cookies.json, but I see that your cookies are not complete.

I just exported the cookies and pasted it into cookie.json.
![image](https://github.com/FuseFairy/DiscordBot-EdgeGPT/assets/78064680/b9d8b2f3-a1b7-4d05-a3d2-1660ffcd9f3b)

**Reply by KreqHex** [reply]
![image](https://github.com/FuseFairy/DiscordBot-EdgeGPT/assets/119231745/eb2a8798-5391-45ff-b759-d4d709c085e0)

this is my cookie now.
still not working.

i tried copy cookie from diffrent machine.

**Reply by FuseFairy** [reply]
can you create image on website?

**Reply by KreqHex** [reply]
the screenshot you send.
i do exactly the same.

**Reply by KreqHex** [reply]
yes i can create image on website did it 1 second ago.

**Reply by FuseFairy** [reply]
but can't find `_U` in cookies, right?
 
![image](https://github.com/FuseFairy/DiscordBot-EdgeGPT/assets/78064680/10dd3b07-e0b1-4af5-bfc4-0b828b244971)

**Reply by KreqHex** [reply]
yes i have the _U right now in my cookie.
(dont added myself)
this is the new cookie from other machine.
so YES i have the _U now.


but still not working...

**Reply by KreqHex** [reply]
![image](https://github.com/FuseFairy/DiscordBot-EdgeGPT/assets/119231745/3d527a07-b15c-405a-b4e4-6c83d5ab5ee8)

**Reply by FuseFairy** [reply]
well, new error?

**Reply by KreqHex** [reply]
Nop same error.
bot gives back:  Error: 'NoneType' object is not iterable
and in console it says:

←[30;1m2024-01-31 13:29:51←[0m ←[34;1mINFO    ←[0m ←[35msrc.image.image_create←[0m -> ←[31mkreqhex←[0m ： 'a sign with a dog' (pingen) [BingImageCreator]
Image create failed pls check cookie or old image still creating
←[30;1m2024-01-31 13:29:55←[0m ←[31mERROR   ←[0m ←[35msrc.image.image_create←[0m -> Error while create image: 'NoneType' object is not iterable

**Reply by FuseFairy** [reply]
Which version of edgegpt are you using?
recommend to use latest

**Reply by KreqHex** [reply]
eh where can i see that?

**Reply by FuseFairy** [reply]
use this version
https://github.com/FuseFairy/DiscordBot-EdgeGPT/releases/tag/v1.1.1

**Reply by KreqHex** [reply]
okay i did.
when u sad wich version are u using i downloaded the newest version xD
but still the same messages.

**Reply by FuseFairy** [reply]
that's strange

**Reply by KreqHex** [reply]
if you want we can go in a discord call and i show you fresh install from the bot so you can see everything i do.

**Reply by KreqHex** [reply]
wich version of python are u using?
maybe thats the problem?

**Reply by FuseFairy** [reply]
ok, but I'll be typing.

ID: zhuang39

**Reply by FuseFairy** [reply]
at least py3.10+

**Reply by FuseFairy** [reply]
or may be you can try replit
https://replit.com/@dd8611706/DiscordBot-EdgeGPT
